eservice/lib/libpdo_enclave/contract_response.cpp

```diff
@@ -132,54 +144,61 @@ ByteArray ContractResponse::SerializeAndEncrypt(const ByteArray& session_key,
-    pdo::error::ThrowIfNull(dependency_array, "failed to serialize the dependency array");
-
-    std::map<std::string, std::string>::const_iterator it;
-    for (it = dependencies_.begin(); it != dependencies_.end(); it++)
-    {
-        JSON_Value* dependency_value = json_value_init_object();
-        pdo::error::ThrowIfNull(dependency_value, "failed to create a dependency array");
+    if (operation_succeeded_) {
+        // --------------- signature ---------------
+        ByteArray signature = ComputeSignature(enclave_data);
+        Base64EncodedString encoded_signature = base64_encode(signature);
 
-        JSON_Object* dependency_object = json_value_get_object(dependency_value);
-        pdo::error::ThrowIfNull(dependency_object, "failed to create a dependency array");
-
-        jret = json_object_dotset_string(dependency_object, "ContractID", it->first.c_str());
+        jret =
+            json_object_dotset_string(contract_response_object, "Signature", encoded_signature.c_str());
         pdo::error::ThrowIf<pdo::error::RuntimeError>(
-            jret != JSONSuccess, "failed to serialize contract id in the dependency list");
+            jret != JSONSuccess, "failed to serialize the signature");
 
-        jret = json_object_dotset_string(dependency_object, "StateHash", it->second.c_str());
+        // --------------- state ---------------
+        Base64EncodedString encoded_state = base64_encode(contract_state_.encrypted_state_);
+        jret = json_object_dotset_string(contract_response_object, "State", encoded_state.c_str());
         pdo::error::ThrowIf<pdo::error::RuntimeError>(
-            jret != JSONSuccess, "failed to serialize contract hash in the dependency list");
+            jret != JSONSuccess, "failed to serialize the state");
 
-        jret = json_array_append_value(dependency_array, dependency_value);
+        // --------------- dependencies ---------------
+        jret = json_object_set_value(contract_response_object, "Dependencies", json_value_init_array());
         pdo::error::ThrowIf<pdo::error::RuntimeError>(
-            jret != JSONSuccess, "failed to add dependency to the serialization array");
+            jret != JSONSuccess, "failed to serialize the dependencies");
+
+        JSON_Array* dependency_array = json_object_get_array(contract_response_object, "Dependencies");
+        pdo::error::ThrowIfNull(dependency_array, "failed to serialize the dependency array");
+
+        std::map<std::string, std::string>::const_iterator it;
+        for (it = dependencies_.begin(); it != dependencies_.end(); it++)
+        {
+            JSON_Value* dependency_value = json_value_init_object();
+            pdo::error::ThrowIfNull(dependency_value, "failed to create a dependency array");
+
+            JSON_Object* dependency_object = json_value_get_object(dependency_value);
+            pdo::error::ThrowIfNull(dependency_object, "failed to create a dependency array");
```

Comment:
This message should be different than the previous one to facilitate debugging. Granted the previous one shouldn't fail... but still.